bin/server.py

```diff
@@ -0,0 +1,181 @@
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+############################################################################
+
+#
+# Script to handle launching the query server process.
+#
+# usage: server.py [start|stop|makeWinServiceDesc] [args]
+#
+
+import datetime
+import getpass
+import os
+import os.path
+import signal
+import subprocess
+import sys
+import fnmatch
+try:
+    import daemon
+    from daemon import pidfile
+    daemon_supported = True
+except ImportError:
+    daemon_supported = False
+
+def getPath():
+    QUARK_SERVER_JAR_PATTERN = "quark-server-*.jar"
+
+    current_dir = os.path.dirname(os.path.abspath(__file__))
+    path = os.path.join(current_dir, "..", "server", "target", "*")
+
+    if ((path is not None) and (len(path) > 0) and (path[-1] == '*')) :
+        path = path[:-1]
+
+    for root, dirs, files in os.walk(path):
+        for name in files:
+            if fnmatch.fnmatch(name, QUARK_SERVER_JAR_PATTERN):
+                return os.path.join(root, name)
+
+
+quark_server_jar = getPath()
+command = None
+args = sys.argv
+
+if len(args) > 1:
+    if args[1] == 'start':
+        command = 'start'
+    elif args[1] == 'stop':
+        command = 'stop'
+    elif args[1] == 'makeWinServiceDesc':
+        command = 'makeWinServiceDesc'
+
+if command:
+    # Pull off server.py and the command
+    args = args[2:]
+else:
+    # Just pull off server.py
+    args = args[1:]
+
+if os.name == 'nt':
+    args = subprocess.list2cmdline(args)
+else:
+    import pipes    # pipes module isn't available on Windows
+    args = " ".join([pipes.quote(v) for v in args])
+
+java_home = os.getenv('JAVA_HOME')
+
+quark_log_dir = os.path.dirname(os.path.abspath(__file__))
+quark_file_basename = 'quark-server-%s' % getpass.getuser()
+quark_log_file = '%s.log' % quark_file_basename
+quark_out_file = '%s.out' % quark_file_basename
+quark_pid_file = '%s.pid' % quark_file_basename
+
+log_file_path = os.path.join(quark_log_dir, quark_log_file)
+out_file_path = os.path.join(quark_log_dir, quark_out_file)
+pid_file_path = os.path.join(quark_log_dir, quark_pid_file)
+
+if java_home:
+    java = os.path.join(java_home, 'bin', 'java')
+else:
+    java = 'java'
+
+# The command is run through subprocess so environment variables are automatically inherited
+java_cmd = '%(java)s -cp ' + \
+           quark_server_jar + \
+           " -Dproc_quarkserver" + \
+           " -Dpsql.root.logger=%(root_logger)s" + \
```

Comment:
The logging parameters will be invalid once you move to logback. So this has to be changed.